components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -1219,23 +1219,38 @@ func (p *pg) ListRulesForProject(ctx context.Context, projectID string) ([]*v2.R
 		}
 	}()
 
+	anyStagedRules := false
 	for rows.Next() {
 		var rule v2.Rule
 		if err := rows.Scan(&rule); err != nil {
-			return nil, p.processError(err)
+			return nil, v2.RulesStatusError, p.processError(err)
+		}
+		if rule.Status == pgStaged {
+			anyStagedRules = true
 		}
 		rules = append(rules, &rule)
 	}
 	if err := rows.Err(); err != nil {
-		return nil, errors.Wrap(err, "error retrieving result rows")
+		return nil, v2.RulesStatusError, errors.Wrap(err, "error retrieving result rows")
 	}
 
 	err = tx.Commit()
 	if err != nil {
-		return nil, storage_errors.NewErrTxCommit(err)
+		return nil, v2.RulesStatusError, storage_errors.NewErrTxCommit(err)
 	}
 
-	return rules, nil
+	var rulesStatus v2.ProjectRulesStatus
+	if len(rules) == 0 {
+		rulesStatus = v2.NoRules
+	} else {
+		if anyStagedRules {
+			rulesStatus = v2.EditsPending
+		} else {
+			rulesStatus = v2.Applied
+		}
+	}
```

Comment:
Let's flatten this:
```go
ruleStatus := v2.Applied
if len(rules) == 0 {
    rulesStatus = v2.NoRules
}
if anyStagedRules {
    ruleStatus = v2.EditsPending
}
```
I guess some similar comment applies to the memstore part. 😉

Reply:
Very similar ;) Sounds like I'm in the vast minority on this one so I'll go ahead and update it in both places!